src/build_system.jl

```diff
@@ -19,100 +17,109 @@ function build_system(
     category::Type{<:SystemCategory},
     name::String,
     print_stat::Bool = false;
+    force_build::Bool = false,
+    assign_new_uuids::Bool = false,
+    skip_serialization::Bool = false,
+    system_catalog::SystemCatalog = SystemCatalog(SYSTEM_CATALOG),
     kwargs...,
 )
-    system_catalog = get(kwargs, :system_catalog, SystemCatalog(SYSTEM_CATALOG))
     sys_descriptor = get_system_descriptor(category, system_catalog, name)
-    return _build_system(category, name, sys_descriptor, print_stat; kwargs...)
-end
+    sys_kwargs = filter_kwargs(; kwargs...)
+    sys_args = Dict{Symbol, Any}(k => v for (k, v) in sys_kwargs)
```

Comment:
Isn’t `sys_kwargs` already a Dict of this type? If not, can `filter_kwargs` be changed to return this type?

Reply:
you are right. This conversion is unnecessary.